workspaces/optic-engine/src/commands/endpoint.rs

```diff
@@ -465,6 +473,17 @@ impl AggregateCommand<EndpointProjection> for EndpointCommand {
         ))]
       }
 
+      EndpointCommand::RemoveRequest(command) => {
+        validation.require(
+          validation.request_exists(&command.request_id),
+          "request id must be in use to be removed",
+        )?;
```

Comment:
I chose to be strict here, where removing a request that doesn't exist fails, but I'm not sure it's necessary. Generally, the outcome the user wanted has been achieved, which is why you see a lot of APIs implemented to be lenient on this. However, it in this case, I imagine we want the user (read: developer issuing this command) to be aware they might be trying to delete something that doesn't exist, and that this might be a bug on their end.

@devdoshi what do you reckon?